gatsby-node.js

```diff
@@ -1,7 +1,62 @@
-/**
- * Implement Gatsby's Node APIs in this file.
- *
- * See: https://www.gatsbyjs.org/docs/node-apis/
- */
+const path = require('path');
+const GalleryPageTemplate = path.resolve('./src/components/GalleryPage/GalleryPageTemplate.js');
 
-// You can delete this file if you're not using it
+function getHothWinners(allHothWinners) {
+	const hothWinners = new Map();
+	const devposts = new Map();
+	for (const hoth of allHothWinners) {
+		const winnersArray = [];
+		for (const winner of hoth.parent.childYaml.winners) {
+			const { category, description, image, link, title } = winner;
+			const winnerMap = { category, description, image, link, title };
+			winnersArray.push(winnerMap);
+		}
+		hothWinners.set(hoth.parent.name, winnersArray);
+		devposts.set(hoth.parent.name, hoth.parent.childYaml.devpost);
+	}
+	return { hothWinners, devposts };
+}
+
+exports.createPages = async ({ actions: { createPage }, graphql }) => {
+	const hothData = await graphql(`
+	query HothDataQuery {
+        allYaml(sort: {fields: name, order: ASC}) {
+            nodes {
+              parent {
+                ... on File {
+                  id
+                  name
+                  childYaml {
+                    name
+                    devpost
+                    winners {
+                      category
+                      description
+                      image
+                      link
+                      title
+                    }
+                  }
+                }
+              }
+            }
+          }
+	  }
+    `);
+	const { hothWinners, devposts } = getHothWinners(hothData.data.allYaml.nodes);
+	for (let i = 1; i <= hothWinners.size; i++) {
+		const hothName = `hoth-${i}`;
+		const winnerInfo = hothWinners.get(hothName);
+		const devpostLink = devposts.get(hothName);
+		const hothNum = i;
```

Comment:
Rather than constructing `hothName` from an index, I'd just loop through all the keys of `hothWinners`, like so:

```js
for (const hothName of hothWinners.keys() {
	const winnerInfo = hothWinners.get(hothName);
	const devpostLink = devposts.get(hothName);
	// ...
```

If you decide to adopt this style, a natural question would then be, what should `hothNum` be set to? For that, I actually suggest that you _remove_ `hothNum`, and change the GalleryPageTemplate and Winners components to accept a `hothName` string instead of `hothNum`. You can then populate `hothName` using the `name` field inside the YAML files. Not only does this proposal allow for more elegant iteration, but (more importantly) it also makes the code more flexible. As an example, it's not unthinkable for future Hack to skip HOTH 9 (like Microsoft Windows and iPhones), or name HOTH 10 as "HOTH X" (like iPhones and macOS). Your current code would be too rigid to handle that case.

Reply:
Thank you, Timothy. I will get right on this.